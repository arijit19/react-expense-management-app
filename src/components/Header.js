import React from 'react';
import {Link,  withRouter} from 'react-router-dom';
import {connect} from 'react-redux';

import { StartLogoutAction} from '../firebase/Actions/LogoutAction.js'
import  { SetAuthAction } from '../Redux/Actions/AuthActions.js'
import {firebase} from '../firebase/firebase.js'

export class Header extends React.Component {

  constructor(props) {
    super(props);
    this.handleLogoutClick = this.handleLogoutClick.bind(this);
    // this.handleAuth();
  }
  
   handleLogoutClick =  async function() {
     console.log(this.props);
     
    const res = await this.props.startLogoutAction();
    if(res !== null){
      this.props.setAuthAction({UID: undefined});
      this.props.history.push('/');
    }
  }

  render() {
    return (
      <header className='header'>
       <div className='content-container'>
         <div className='header__content'>
            {/* <p><NavLink exact activeClassName='active-class' to='/' >Home Page</NavLink></p> */}
            <Link to='/dashboard' className='header__title'>
              <h1>Expense Tracker</h1>
            </Link>
            <button onClick={this.handleLogoutClick} className='button button--link'>Logout</button>
         </div>
       </div>
      </header>
  );
  }
}

const mapDispatchToProps = (dispatch)=> {
  return {
    startLogoutAction: ()=> dispatch(StartLogoutAction()),
    setAuthAction: (data)=> dispatch(SetAuthAction(data))
  }
}


export default connect(undefined,mapDispatchToProps) (withRouter(Header));