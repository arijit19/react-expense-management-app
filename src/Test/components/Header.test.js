import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../components/Header.js';

test('Testing Header renderer',()=>{
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
});