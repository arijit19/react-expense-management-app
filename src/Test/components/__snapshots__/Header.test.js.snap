// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header renderer 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Expense Tracker
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
